rs/replicated_state/src/canister_state/system_state/call_context_manager.rs

```diff
@@ -784,12 +797,34 @@ impl CallContextManager {
     pub fn unregister_callback(&mut self, callback_id: CallbackId) -> Option<Arc<Callback>> {
         self.callbacks.remove(&callback_id).map(|callback| {
             self.stats.on_unregister_callback(&callback);
+            if callback.deadline != NO_DEADLINE {
+                self.unexpired_callbacks
+                    .remove(&(callback.deadline, callback_id));
+            }
             debug_assert!(self.stats_ok());
 
             callback
         })
     }
 
+    /// Returns the IDs of all best-effort callbacks whose deadlines have expired
+    /// since the previous call to this method, given the current time.
+    ///
+    /// Note: A given callback will be returned at most once by this function.
+    #[allow(dead_code)]
+    pub(super) fn expired_callbacks(
+        &mut self,
+        now: CoarseTime,
+    ) -> impl Iterator<Item = CallbackId> {
+        let mut expired_callbacks = self
+            .unexpired_callbacks
+            .split_off(&(now, CallbackId::from(0)));
```

Comment:
I suppose we use `CallbackId::from(0)` here because it is `<=` all valid callback ids, right? Maybe worth a comment.

Reply:
Defined a `MIN_CALLBACK_ID` constant as documentation. So it now says `split_off(&(now, MIN_CALLBACK_ID))`.